routes/auth.js

```diff
@@ -0,0 +1,85 @@
+var express = require('express')
+  , User = require('../models/user')
+  , passport = require('passport')
+  , router = express.Router();
+
+router.get('/', function(req, res) {
+  if(req.session.passport.user === undefined) {
+    res.redirect('/login');
+  } else {
+    res.redirect('/profile');
+  }
+});
+
+router.get('/login', function(req, res) {
+  if(req.session.passport.user === undefined) {
+    res.render('login', { csrfToken: req.csrfToken() });
+  } else {
+    res.redirect('/profile');
+  }
+});
+
+router.post('/login', function(req, res, next) {
+  // TODO: Implement 3-time failure restriction
+  // and check if it's a locked or banned account
+  // We should probably do it here, before going
+  // to the authentication. Otherwise, we can look
+  // into the library source code and customize it.
+  // Also, a new field, number of failures, may be
+  // added to the User model.
```

Comment:
`passport-local-mongoose` handles some of this already. (See comment on `config/passport.js`)